src/main/java/uk/gov/justice/digital/delius/service/CourtService.java

```diff
@@ -116,9 +134,41 @@ public Court getCourt(String code) {
             .toList();
     }
 
-    public static record CourtAlreadyExists(String courtCode) {
+    public interface BadData {
+        String message();
+    }
+
+    public interface NotFound extends BadData {
+    }
+
+    public static record CourtAlreadyExists(String courtCode) implements BadData {
+        public String message() {
+            return String.format("Court %s already exists", courtCode);
+        }
+    }
+
+    public static record CourtDoesNotExist(String courtCode) implements NotFound {
+        public String message() {
+            return String.format("Court %s does not exist", courtCode);
+        }
+    }
+
+    public static record CourtTypeDoesNotExist(String courtTypeCourt) implements BadData {
+        public String message() {
+            return String.format("Court type %s does not exist", courtTypeCourt);
+        }
     }
 
-    public static record CourtDoesNotExist(String courtCode) {
+    public static record ProbationDoesNotExist(String probationAreaCode) implements BadData {
+        public String message() {
+            return String.format("Probation area %s does not exist", probationAreaCode);
+        }
+    }
+
```

Comment:
small helper to give clarity to the mean of left and right when using Either library

Reply:
more readable